crypto-shared/src/kdf.rs

```diff
@@ -0,0 +1,89 @@
+use crate::types::{PublicKey, ScalarExt};
+use anyhow::Context;
+use k256::{
+    ecdsa::{RecoveryId, Signature, VerifyingKey},
+    elliptic_curve::{point::AffineCoordinates, sec1::ToEncodedPoint, CurveArithmetic},
+    sha2::{Digest, Sha256},
+    Scalar, Secp256k1,
+};
+use near_account_id::AccountId;
+
+// Constant prefix that ensures epsilon derivation values are used specifically for
+// near-mpc-recovery with key derivation protocol vX.Y.Z.
+const EPSILON_DERIVATION_PREFIX: &str = "near-mpc-recovery v0.1.0 epsilon derivation:";
+
+pub fn derive_epsilon(predecessor_id: &AccountId, path: &str) -> Scalar {
+    // TODO: Use a key derivation library instead of doing this manually.
+    // https://crates.io/crates/hkdf might be a good option?
+    //
+    // ',' is ACCOUNT_DATA_SEPARATOR from nearcore that indicate the end
+    // of the accound id in the trie key. We reuse the same constant to
+    // indicate the end of the account id in derivation path.
+    let derivation_path = format!("{EPSILON_DERIVATION_PREFIX}{},{}", predecessor_id, path);
+    let mut hasher = Sha256::new();
+    hasher.update(derivation_path);
+    let mut bytes = hasher.finalize();
```

Comment:
consider native [near_sdk::env::sha256](https://docs.rs/near-sdk/latest/near_sdk/env/fn.sha256.html) instead of wasm-side sha256 to be more efficient?

Reply:
Will do